tests/implement/tests/vector_view.rs

```diff
@@ -0,0 +1,141 @@
+use std::vec;
+use test_implement::*;
+use windows::*;
+use Windows::Foundation::Collections::*;
+
+#[implement(
+    Windows::Foundation::Collections::IVectorView<T>,
+)]
+struct TestView<T>(Vec<T>)
+where 
+    T: ::windows::RuntimeType + 'static;
+
+#[allow(non_snake_case)]
+impl<T: ::windows::RuntimeType + 'static> TestView<T> {
+    /* GetAt returns the value at the given index, and panic if given an invalid index */
+    fn GetAt(&self, index : u32) -> Result<T> {
+        let vec_size = self.0.len() as u32;
+        assert!(index < vec_size, "Given index ({}) was out of bounds for the IVectorView (length {})", index, vec_size);
```

Comment:
Since this is intended for library use, we can't have the implementation panic. Instead, this should return `E_BOUNDS`. [Here's an example](https://github.com/microsoft/windows-rs/blob/6116afb9e8c426dd90ca44e654aede9208e5400f/tests/implement/tests/into_impl.rs#L25).